src/client/datascience/notebook/helpers/helpers.ts

```diff
@@ -346,13 +330,39 @@ function createNotebookCellDataFromCodeCell(cell: nbformat.ICodeCell, cellLangua
 
     const source = concatMultilineString(cell.source);
 
-    return {
-        cellKind: NotebookCellKind.Code,
-        language: cellLanguage,
-        metadata: notebookCellMetadata,
-        source,
-        outputs
-    };
+    return new NotebookCellData(NotebookCellKind.Code, source, cellLanguage, outputs, notebookCellMetadata);
+}
+const orderOfMimeTypes = [
+    'application/vnd.*',
+    'application/vdom.*',
+    'application/geo+json',
+    'application/x-nteract-model-debug+json',
+    'text/html',
+    'application/javascript',
+    'image/gif',
+    'text/latex',
+    'text/markdown',
+    'image/svg+xml',
+    'image/png',
+    'image/jpeg',
+    'application/json',
+    'text/plain'
+];
+function sortOutputItemsBasedOnDisplayOrder(outputItems: NotebookCellOutputItem[]): NotebookCellOutputItem[] {
+    return outputItems.sort((outputItemA, outputItemB) => {
+        const isMimeTypeMatch = (value: string, compareWith: string) => {
+            if (value.endsWith('.*')) {
+                value = value.substr(0, value.indexOf('.*'));
+            }
+            return compareWith.startsWith(value);
+        };
+        const indexOfMimeTypeA = orderOfMimeTypes.findIndex((mime) => isMimeTypeMatch(outputItemA.mime, mime));
+        const indexOfMimeTypeB = orderOfMimeTypes.findIndex((mime) => isMimeTypeMatch(outputItemB.mime, mime));
+        if (indexOfMimeTypeA === indexOfMimeTypeB) {
```

Comment:
I believe you could just return 'indexOfMimeTypeA' - 'indexOfMimeTypeB'

Reply:
lol, yes